### DISCUSSION TITLE
trf French model and NER

### DISCUSSION THREAD

**Discussion by theognis0110** [explosion/spaCy#8030] [Help: Best practices]
Hi,

the **fr_dep_news_trf** model has  a better accuracy for dependency parsing but on the other hand it doesn't include a NER. Is it mandatory to use a different model such as **fr_core_news_lg** to get a NER ?
i attempted to add a ner to the trf but it doesn't seem to be possible or maybe i miss something about it ?

another question about the trf : i am only interested in the tagger and the dependency parse results; in case i can combine the trf model with a ner,  can i just remove the transformer from the pipeline ?  (at this point  i don't intend to train anything)

**Comment by polm** [comment]
The components in the pretrained pipelines rely on either a tok2vec or transformer component to tranform the raw input into vector representations. The components are conditioned on the upstream component they come with, so you can't just swap that out. 

> i attempted to add a ner to the trf but it doesn't seem to be possible or maybe i miss something about it ?

You can't copy an NER component from another pipeline and have it work as-is unless you also include the tok2vec it relies on. See the section on [sharing embedding layers](https://spacy.io/usage/embeddings-transformers#embedding-layers) in the docs for some more info on this. 

> can i just remove the transformer from the pipeline ? 

No, the transformer creates the input for the other components in the trf pipeline, so it's required.
